app/src/components/common/TestURLModal/index.tsx

```diff
@@ -0,0 +1,228 @@
+  }, [analyticsContext, source.operator]);
+
+  useEffect(() => {
+    if (isSourceModified) {
+      trackURLConditionSourceModified(analyticsContext, source.operator);
+    }
+  }, [analyticsContext, source.operator, isSourceModified]);
+
+  useEffect(() => {
+    if (isTestURLTried) {
+      trackURLConditionMatchingTried(analyticsContext, source.operator);
+    }
+  }, [isTestURLTried, analyticsContext, source.operator]);
+
+  return (
+    <RQModal
+      centered
+      open={isOpen}
+      className="test-url-modal"
+      width={800}
+      onCancel={() => onClose(sourceConfig.operator)}
+    >
+      <div className="test-url-modal-header">
+        <Typography.Title level={4}>Test URL condition</Typography.Title>
+        <Typography.Text className="text-gray">
+          Check if your request URL matches the rule condition you specified.
+        </Typography.Text>
+      </div>
+      <Divider />
+      <div className="test-url-modal-body">
+        <div className="text-bold white">Source condition</div>
+        <div className="source-condition-input-wrapper mt-8">
+          <Col className="shrink-0">
+            <Select
+              value={sourceConfig.key}
+              className="source-condition-selector cursor-pointer uppercase"
+              onChange={(value) => {
+                handleSourceConfigChange("key", value);
+                handleTestURL(testURL, null, null, value);
+              }}
+            >
+              {Object.entries(SourceKey).map(([key, value]) => (
+                <Select.Option key={value} value={value}>
+                  {capitalize(value)}
+                </Select.Option>
+              ))}
+            </Select>
+          </Col>
+          <Col className="shrink-0">
+            <Select
+              value={sourceConfig.operator}
+              className="source-condition-selector cursor-pointer"
+              onChange={(value) => {
+                handleSourceConfigChange("operator", value);
+                handleTestURL(testURL, null, value, null);
+              }}
+            >
+              {Object.entries(SourceOperator).map(([key, value]) => (
+                <Select.Option key={key} value={value}>
+                  {value === SourceOperator.WILDCARD_MATCHES
+                    ? "Wildcard"
+                    : value === SourceOperator.MATCHES
+                    ? "RegEx"
+                    : value}
+                </Select.Option>
+              ))}
+            </Select>
+          </Col>
+          <Input
+            className="source-url-input"
+            placeholder="Enter source URL"
+            value={sourceConfig.value}
+            ref={sourceInputRef}
+            onChange={handleSourceInputChange}
+          />
+          {sourceConfig.operator === SourceOperator.MATCHES && !isRegexFormat(sourceConfig.value) && (
+            <div className="invalid-regex-badge">INVALID REGEX</div>
+          )}
+        </div>
```

Comment:
This should be abstracted as a common component. We can use this component in Rule Editor, Session Recording Config etc. Rest of the logic to update the source value in their respective storage can be handled by the parent component

```
interface Props {
    onChange(updatedSource) => void
}

UrlSource ({Props}) => {}
```